lib/util/createHash.js

```diff
@@ -4,17 +4,41 @@
 */
 "use strict";
 
-/** @typedef {{new(): Hash}} HashConstructor */
-/**
- * @typedef {Object} Hash
- * @property {function(string|Buffer, string=): Hash} update
- * @property {function(string): string} digest
- */
-
 const BULK_SIZE = 1000;
 
-class BulkUpdateDecorator {
+class BaseHash {
+	/**
+	 * Update hash {@link https://nodejs.org/api/crypto.html#crypto_hash_update_data_inputencoding}
+	 * @param {string|Buffer} data data
+	 * @param {string} [inputEncoding] data encoding
+	 * @returns {BaseHash} updated hash
+	 */
+	update(data, inputEncoding) {
+		return this;
+	}
+
+	/**
+	 * Calculates the digest {@link https://nodejs.org/api/crypto.html#crypto_hash_digest_encoding}
+	 * @param {string} [encoding] encoding of the return value
+	 * @returns {string|Buffer} digest
+	 */
+	digest(encoding) {
+		return "";
+	}
+}
+
+/** @typedef {BaseHash} Hash */
+/** @typedef {{new(): BaseHash}} HashConstructor */
+
+/**
+ * @extends {BaseHash}
```

Comment:
Typescript currently does not support `@implements`

Reply:
Is that `@extends {BaseHash}` needed when the class `extends BaseHash`?

Reply:
In this case no, where it matter - https://www.typescriptlang.org/docs/handbook/type-checking-javascript-files.html#in-extends-clause